ext/drbbdup/drbbdup.c

```diff
@@ -735,14 +743,15 @@ drbbdup_encode_runtime_case(void *drcontext, drbbdup_per_thread *pt, void *tag,
                                DRBBDUP_SCRATCH_REG);
     }
 #ifdef AARCHXX
-    drreg_is_register_dead(drcontext, DRBBDUP_SCRATCH_REG2, where,
-                           &manager->is_scratch_reg2_dead);
-    if (!manager->is_scratch_reg2_dead) {
-        /* TODO i#4134: Add a user-set flag promising that runtime case values are
-         * always <256 so we know we don't need this 2nd scratch register.
-         */
-        drbbdup_spill_register(drcontext, bb, where, DRBBDUP_SCRATCH_REG2_SLOT,
-                               DRBBDUP_SCRATCH_REG2);
+    if (opts.max_case_encoding > 0 && opts.max_case_encoding < MAX_IMMED_IN_CMP)
+        manager->is_scratch_reg2_dead = true; /* Avoid restore, since not used. */
```

Comment:
I personally would have introduced a new flag that denotes whether reg2 is in use rather that piggybacking on the dead flag. It makes the code more readable and maybe less error-prone  as now the dead flag can be true when it actual fact the register is not.

Reply:
Done.